src/ros_filter.cpp

```diff
@@ -133,7 +135,10 @@ void RosFilter<T>::reset()
 
   // Also set the last set pose time, so we ignore all messages
   // that occur before it
-  last_set_pose_time_ = rclcpp::Time(0);
+  last_set_pose_time_ = rclcpp::Time(0, 0, RCL_ROS_TIME);
```

Comment:
Do you really want to set the clock? This would make it so that you always use the ROS clock and not the other options.

Reply:
If you launched a test with `use_sim_time` it would use ROS time

Reply:
If you don't set it, it defaults to `RCL_SYSTEM_TIME`, right? And if I don't set this, the code throws an exception, because we end up comparing the current message time stamp (which is in ROS time) with this, which is in system time.

Reply:
Wouldn't the tests automatically do that?

Also, is this not like ROS1? Does ROS time not == system time for a live system, and sim time in bags?

Reply:
Just linking this so we're talking about the same stuff. You obviously have been working with ROS2 a lot more than I have, so apologies if I have this wrong.

https://design.ros2.org/articles/clock_and_time.html

> The ROSTime will report the same as SystemTime when a ROS Time Source is not active. When the ROS time source is active ROSTime will return the latest value reported by the Time Source. ROSTime is considered active when the parameter use_sim_time is set on the node.

This is just like ROS1. The problem here is that if you _don't_ provide constructor arguments, you'll get the second constructor signature here:

https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Time.html
https://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Time.html#a683d5111f099d16532f10fcd82e8f5af

That means an uninitialized `rclcpp::Time` object will default to `SYSTEM_TIME`, as I read it. And that was the issue: the new `last_published_stamp_` was not being initialized with a different constructor, so it was defaulting to `RCL_SYSTEM_TIME`, but we later compared it to a message stamp, which is obviously going to be of type `RCL_ROS_TIME`. That throws an exception, causing all my tests to fail.

In any case, tests are now all passing. 🥳